internal/local/structure.go

```diff
@@ -230,3 +232,118 @@ func parseServices(rootDir string) ([]ServiceStructure, error) {
 
 	return out, nil
 }
+
+func writeSecrets(rootDir string, secrets *SecretsStructure) error {
+	if secrets == nil {
+		return nil
+	}
+	data, err := MarshalJSON(secrets)
+	if err != nil {
+		return err
+	}
+	return WriteFile(filepath.Join(rootDir, FileSecrets.String()), 0666, bytes.NewReader(data))
+}
+
+func writeEnvironments(rootDir string, environments map[string]map[string]interface{}) error {
+	for env, values := range environments {
+		data, err := MarshalJSON(values)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameEnvironments, env), 0666, bytes.NewReader(data))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeValues(rootDir string, values []map[string]interface{}) error {
+	if err := os.MkdirAll(filepath.Join(rootDir, NameValues), os.ModePerm); err != nil {
+		return err
+	}
+	for _, vals := range values {
+		data, err := MarshalJSON(vals)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameValues, fmt.Sprintf("%s%s", vals["name"], extJSON)), 0666, bytes.NewReader(data))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeGraphQL(rootDir string, graphql GraphQLStructure) error {
+	if err := os.MkdirAll(filepath.Join(rootDir, NameGraphQL, NameCustomResolvers), os.ModePerm); err != nil {
+		return err
+	}
+	if graphql.Config != nil {
+		data, err := MarshalJSON(graphql.Config)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameGraphQL, FileConfig.String()), 0666, bytes.NewReader(data))
+		if err != nil {
+			return err
+		}
+	}
+	for _, vals := range graphql.CustomResolvers {
+		data, err := MarshalJSON(vals)
+		if err != nil {
+			return err
+		}
+		nameFile := fmt.Sprintf("%s_%s%s", strings.ToLower(vals["on_type"].(string)), strings.ToLower(vals["field_name"].(string)), extJSON)
+		err = WriteFile(filepath.Join(rootDir, NameGraphQL, NameCustomResolvers, nameFile), 0666, bytes.NewReader(data))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeServices(rootDir string, services []ServiceStructure) error {
+	if err := os.MkdirAll(filepath.Join(rootDir, NameServices), os.ModePerm); err != nil {
+		return err
+	}
+	for _, svc := range services {
+		nameService := svc.Config["name"].(string)
+		data, err := MarshalJSON(svc.Config)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameServices, nameService, FileConfig.String()), 0666, bytes.NewReader(data))
+		if err != nil {
+			return err
+		}
+		for _, wh := range svc.IncomingWebhooks {
+			dataSrc := wh[NameSource].(string)
+			nameWebhook := wh["name"].(string)
+			delete(wh, NameSource)
```

Comment:
server code looks like we still write the `name` field, any reason this is deleted here?

Reply:
This is deleting the source code from the map, so that it doesn't get thrown into the config file.

Reply:
gotcha, i read that wrong at first and thought we were deleting the `"name"` field 🙄 

as with the other cases of this, lets just make sure the `AppData` stays in tact after this `delete`

Reply:
Yeah, I just did a test and it doesn't... I thought the value loop in go created a separate copy. So I'm creating temp copies in my next commit.